go/test/endtoend/transaction/twopc/main_test.go

```diff
@@ -126,6 +127,26 @@ func cleanup(t *testing.T) {
 	defer conn.Close()
 
 	_, _ = utils.ExecAllowError(t, conn, "delete from twopc_user")
+	clearOutTable(t, conn, "twopc_fuzzer_insert")
+	clearOutTable(t, conn, "twopc_fuzzer_update")
+}
+
+// clearOutTable deletes everything from a table. Sometimes the table might have more rows than allowed in a single delete query,
+// so we have to do the deletions iteratively.
+func clearOutTable(t *testing.T, conn *mysql.Conn, tableName string) {
+	for {
+		res, err := conn.ExecuteFetch(fmt.Sprintf("SELECT count(*) FROM %v", tableName), 1, false)
+		require.NoError(t, err)
+		require.Len(t, res.Rows, 1)
+		require.Len(t, res.Rows[0], 1)
+		rowCount, err := res.Rows[0][0].ToInt()
+		require.NoError(t, err)
+		if rowCount == 0 {
+			return
+		}
+		_, err = conn.ExecuteFetch(fmt.Sprintf("DELETE FROM %v LIMIT 10000", tableName), 10000, false)
+		require.NoError(t, err)
+	}
 }
```

Comment:
this is a select followed by delete on vtgate.
We can do shard targeting and then executing delete with limit.

Reply:
I don't think we need to optimize deletes to that level, this is just to clear the database.

Reply:
yes, but internally it will again use distributed commit. 
I am fine with this, just stating it out.

Reply:
I thought of that too, but then decided it doesn't matter. This entire fuzzer is meant for atomic distributed transactions. This too should always work.